lib/common-types.ts

```diff
@@ -1050,7 +1060,16 @@ export type MRReturn<
   Action extends keyof MRActions[ET]
 > = 'return' extends keyof MRActions[ET][Action] ? Promise<MRActions[ET][Action]['return']> : never
 
-export interface MakeRequestOptions {
+interface MakeRequestOptionsPatch {
+  entityType: keyof MRActions
+  action: 'patch'
+  params?: Record<string, unknown>
+  payload: OpPatch[]
+  headers?: Record<string, unknown>
+  userAgent: string
+}
+
+interface MakeRequestOptionsNonPatch {
```

Comment:
I think you can merge those. `MakeRequestOptions` is not very accurate anyway. E.g. `payload` is allowed for every action even though get- and delete-actions do not accept payloads.